code/modules/atmospherics/machinery/portable/canister.dm

```diff
@@ -428,6 +428,7 @@ GLOBAL_LIST_INIT(gas_id_to_canister, init_gas_id_to_canister())
 
 /obj/machinery/portable_atmospherics/canister/deconstruct(disassembled = TRUE)
 	if((flags_1 & NODECONSTRUCT_1))
+		qdel(src)
```

Comment:
wat the fuck

Reply:
When I was combing through deconstruct overrides I noticed that basically every single one qdels the object whether or not it has the NODECONSTRUCT_1 flag. Generally the only thing that flag does for most objects is change whether they drop items when deconstructed or not. The logic for gas canisters, as it was, meant that if a canister had the NODECONSTRUCT flag, it would return before being destroyed. This was inconsistent with all other objects, and would make it impossible for a player to remove a (hypothetical) holodeck canister, so I added qdel in to that case manually.

Reply:
why is it called no_deconstruct if it doesn't block deconstruction, what the hell

Reply:
gah you're right hhhhhhhhhhh

Reply:
honestly i could rename the flag -- NO_ITEM_ON_DECONSTRUCT or something

Reply:
it's,,, ok. We can do this